File: Coding In Flow/Services And Background Tasks/IntentService/app/src/main/java/com/example/intentservice/MainActivity.java
package com.example.intentservice;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.app.IntentService;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private EditText edtInput;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtInput = findViewById(R.id.edt_input);
    }

    public void startService(View view) {
        String input = edtInput.getText().toString();
        Intent serviceIntent = new Intent(this, ExampleIntentService.class);
        serviceIntent.putExtra("inputExtra", input);

        ContextCompat.startForegroundService(this, serviceIntent);
    }
}

File: Personal/MessengerDemo/messenger_service/src/main/java/com/vanphuc/messenger_service/MainActivity.kt
package com.vanphuc.messenger_service

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var serviceIntent: Intent? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClickStartService(view: View) {
        serviceIntent = Intent(applicationContext, RemoteService::class.java)
        startService(serviceIntent)
    }

    fun onClickStopService(view: View) {
        stopService(serviceIntent)
    }
}